### PR TITLE
Refactor: lib/services (just a first batch)

### PR DESCRIPTION
**Pull request by jnpkrn**

### CODE REVIEW

File: lib/services/systemd.h [hunk 1]
```diff
@@ -16,8 +16,13 @@
...
+#ifndef SYSTEMD__H
+#  define SYSTEMD__H
+
 G_GNUC_INTERNAL GList *systemd_unit_listall(void);
 G_GNUC_INTERNAL int systemd_unit_exec(svc_action_t * op);
 G_GNUC_INTERNAL gboolean systemd_unit_exists(const gchar * name);
 G_GNUC_INTERNAL gboolean systemd_unit_running(const gchar * name);
 G_GNUC_INTERNAL void systemd_cleanup(void);
+
+#endif  /* SYSTEMD__H */
```
Comment (kgaillot) [inline lib/services/systemd.h]: We should use SERVICES_SYSTEMD_H here; SYSTEMD_H seems more likely to have a conflict.

File: lib/services/upstart.h [hunk 2]
```diff
@@ -29,4 +28,4 @@ G_GNUC_INTERNAL gboolean upstart_job_exists(const gchar * name);
 G_GNUC_INTERNAL gboolean upstart_job_running(const gchar * name);
 G_GNUC_INTERNAL void upstart_cleanup(void);
 
-#endif                          /* _UPSTART_DBUS_H_ */
+#endif  /* UPSTART__H */
```
Comment (kgaillot) [inline lib/services/upstart.h]: Ditto here

File: lib/services/dbus.c [hunk 3]
```diff
@@ -8,7 +8,6 @@
 #include <crm_internal.h>
 #include <crm/crm.h>
 #include <crm/services.h>
-#include <dbus/dbus.h>
```
Comment (kgaillot) [inline lib/services/dbus.c]: I think it's good practice to leave this in, since this file requires it, independently of pcmk-dbus.h's internal need for it.

File: lib/services/upstart.h [hunk 4]
```diff
@@ -23,7 +23,7 @@
 #  include "crm/services.h"
 
 G_GNUC_INTERNAL GList *upstart_job_listall(void);
-G_GNUC_INTERNAL int upstart_job_exec(svc_action_t * op, gboolean synchronous);
+G_GNUC_INTERNAL gboolean upstart_job_exec(svc_action_t * op);
```
Comment (kgaillot) [inline lib/services/upstart.h]: Ideally removing synchronous would be done in the relevant commit

File: lib/services/pcmk-dbus.h [hunk 5]
```diff
@@ -14,19 +14,32 @@
 #    define DBUS_TIMEOUT_USE_DEFAULT -1
 #  endif
 
+G_GNUC_INTERNAL
```
Comment (kgaillot) [inline lib/services/pcmk-dbus.h]: Do we define G_GNUC_INTERNAL anywhere if we're not using gcc? Perhaps we should define our own symbol, that evaluates to G_GNUC_INTERNAL if gcc is in use, and empty otherwise.

File: lib/services/upstart.h (hunk 4 above)
Comment (jnpkrn) [inline lib/services/upstart.h]: Ah, you can see how I like this cut-several-changes-back-to-commits process :-)
I've missed this.

File: lib/services/systemd.h (hunk 1 above)
Comment (jnpkrn) [inline lib/services/systemd.h]: Note that itintentionally, per pre-existing convention, ends with `__H`.
As such, I've verified that there is no clash (together with `UPSTART__H`
across the full codebase of several distributions.

We should unify the namespacing of the guard names, but I will send that
as a separate commit/PR.

File: lib/services/dbus.c (hunk 3 above)
Comment (jnpkrn) [inline lib/services/dbus.c]: Ok, there are direct `dbus.h` uses, indeed.

**Comment by jnpkrn** [comment]
Also added 54068af and made `action_exec_helper` explicitly inline.

File: lib/services/pcmk-dbus.h (hunk 5 above)
Comment (jnpkrn) [inline lib/services/pcmk-dbus.h]: I left it out as a business of `glib` to do the right thing.  Apparently
there are prerequisites that it can use (gcc reveals itself to the code
via some extra macros).  And indeed that's the case (it would be sad if
not, it would mean that pacemaker code was already broken before).

File: lib/services/systemd.h (hunk 1 above)
Comment (kgaillot) [inline lib/services/systemd.h]: Certainly it's good that there are no conflicts now, but I meant that SYSTEMD__H seems like the kind of name that someone else might easily use in the future, causing us sudden strange build errors. I just think it's better to make it less generic -- SERVICES_SYSTEMD__H is consistent with other headers we use, and less likely to be used by someone else. It would be even better if we consistently used a unique prefix like CRMD_ or PCMK_, but I think SERVICES_SYSTEMD__H is unlikely enough.

File: lib/services/pcmk-dbus.h (hunk 5 above)
Comment (kgaillot) [inline lib/services/pcmk-dbus.h]: Ah, I didn't realize that was glib symbol and not a gcc symbol.

File: lib/services/systemd.h (hunk 1 above)
Comment (jnpkrn) [inline lib/services/systemd.h]: Yes, these and all other cases will be covered with a separate PR I am currently working on.  You can postpone this one until the other/global one is ready.

Comment (jnpkrn) [inline lib/services/systemd.h]: See #1224.
